napari/_qt/dialogs/qt_reader_dialog.py

```diff
@@ -34,9 +34,16 @@ def __init__(
         self.setWindowTitle(trans._('Choose reader'))
         self._current_file = pth
 
-        if os.path.isdir(pth) and str(pth).endswith('/'):
-            pth = os.path.dirname(pth)
         self._extension = os.path.splitext(pth)[1]
+        self._persist_text = f'Remember this choice for files with a {self._extension} extension'
+
+        if os.path.isdir(pth):
+            self._extension = os.path.basename(pth)
+            if not self._extension.endswith(
+                '.zarr'
+            ) and not self._extension.endswith('/'):
+                self._extension = self._extension + '/'
```

Comment:
We probably want to also add a leading `/` here to the extension. If we want to save just on the directory name (rather than full path), we probably want to save `*/my-folder/`. If we just save `*my-folder/` as this is currently doing, we'll catch stuff like `/other-words-my-folder/`

Reply:
`self._extension` will start with `.` and I as I correctly understand in will produce `.my_folder_ext/` and will apply to `a.my_folder_ext` and `b.my_folder_ext` etc. 

Also, this part of the code should be tested as different systems handle paths differently. 

```suggestion
                self._extension = self._extension + os.sep
```

https://docs.python.org/3/library/os.html#os.sep